### ISSUE TITLE
Deleting Attribute Set products deleted

### ISSUE THREAD

**Issue by TangLiang** [OpenMage/magento-lts#4360]
### Steps to reproduce (*)
1.Make a new attribute set
2.Assign attribute set to product
3.delete attribute set

### Actual result (*)
The product gets deleted from the DB and everywhere.

**Comment by justinbeaty** [comment]
The admin panel should give a warning about this, no? 

https://github.com/OpenMage/magento-lts/blob/7806f0e529da17a85df6f82c09420116a1a60f26/app/code/core/Mage/Adminhtml/Block/Catalog/Product/Attribute/Set/Main.php#L95-L105

Would it be better to not allow deletion of the attribute set if products are assigned to it? Or require the user to type something like “confirm deletion” into a box to proceed?

I believe it’s this way because I’m not sure changing the attribute set to the default one is possible since there could be required fields present in the default set that weren’t in the deleted set.

**Comment by TangLiang** [comment]
i think it's much better to reassign the product to default with basic attributes

**Comment by sreichel** [comment]
> The admin panel should give a warning about this, no?

Yes. But ... its stupid to delete products base on attribute-set.

Better add code from https://github.com/flagbit/Magento-ChangeAttributeSet as forked module in OM namespace?

**Comment by justinbeaty** [comment]
@sreichel Yes that would be nice, but that module is GPL... Can we get all contributors of that module to agree to relicense under OSL?

**Comment by sreichel** [comment]
GPL2 (not)3 shoulb be okay (???). I dont know if it makes i difference, but most of current code is written/changed by me.

**Comment by justinbeaty** [comment]
> GPL2 (not)3 shoulb be okay (???). I dont know if it makes a difference, but most of current code is written/changed by me.

I guess that if it stays as a separate repo it's fine as GPL, but to integrate into core I don't think GPLv2 is compatible either. There are not many contributors to the repo it seems and some of the old contributions might have been replaced by new code by now. So I don't think it would be too hard to relicense if everyone is willing, or to use only some parts of the code (i.e. parts you wrote) and replace others.

**Comment by addison74** [comment]
Deleting an attribute set is part of OpenMage and this feature was established by the Magento team long time in the past. In my opinion it works correctly so I will move the post to the Discussions section.
